src/main/java/org/springframework/data/couchbase/core/query/QueryCriteria.java

```diff
@@ -52,77 +66,177 @@ public QueryCriteria or(String key) {
 		return new QueryCriteria(this.criteriaChain, key, null, ChainOperator.OR);
 	}
 
+	public QueryCriteria child(QueryCriteria criteria) {
+		QueryCriteria qc= new QueryCriteria(null, criteria.key, criteria.value,null, criteria.operator, criteria.format);
+		return qc;
+	}
+
 	public QueryCriteria eq(@Nullable Object o) {
 		return is(o);
 	}
 
 	public QueryCriteria is(@Nullable Object o) {
 		operator = "=";
-		value = o;
+		value = new Object[]{o};
 		return this;
 	}
 
 	public QueryCriteria ne(@Nullable Object o) {
 		operator = "!=";
-		value = o;
+		value = new Object[]{o};
 		return this;
 	}
 
 	public QueryCriteria lt(@Nullable Object o) {
 		operator = "<";
-		value = o;
+		value = new Object[]{o};
 		return this;
 	}
 
 	public QueryCriteria lte(@Nullable Object o) {
 		operator = "<=";
-		value = o;
+		value = new Object[]{o};
 		return this;
 	}
 
 	public QueryCriteria gt(@Nullable Object o) {
 		operator = ">";
-		value = o;
+		value = new Object[]{o};
 		return this;
 	}
 
 	public QueryCriteria gte(@Nullable Object o) {
 		operator = ">=";
-		value = o;
+		value = new Object[]{o};
+		return this;
+	}
+
+	public QueryCriteria startingWith(@Nullable Object o) {
+		regex("^"+o.toString());
```

Comment:
wouldn't that be more the LIKE operator?

Reply:
LIKE '<value>%' should also work.